libyul/backends/wasm/polyfill/Interface.yul

```diff
@@ -51,7 +51,8 @@ function caller() -> z1, z2, z3, z4 {
 
 function callvalue() -> z1, z2, z3, z4 {
 	eth.getCallValue(0:i32)
-	z1, z2, z3, z4 := mload_internal(0:i32)
+	z3 := i64.load(8:i32)
+	z4 := i64.load(0:i32)
 }
```

Comment:
There is no `z_1` and `z_2`. Also this became `z3 := i64.load(...)` instead of multiple. Both `UnusedFunctionPruner` (since `z_1` and `z_2` are unused now) and `ExpressionInliner` (for things that look like `let x := f()`) would be applicable now. @cameel would that be relevant for the change in json? Not sure.

Similarly, also for `gasprice`.

Reply:
Oh, so an expression of the form `z1, z2, z3, z4 := mload_internal(0:i32)` could prevent the inliner from inlining stuff?

If so, then this explains the difference.

Reply:
There are two inliners, `FullInliner` and `ExpressionInliner`.

`ExpressionInliner` only works with expressions of the form `let x := f(...)`.  or `x := f(...)`, i.e., single return variable. Whereas `FullInliner` is more general. It could be that, the change allows `ExpressionInliner` to kick in. However, this is not so easy to check.

Reply:
OK. Just the fact that this is possible is enough for me. I orignally thought this change was very suspect and maybe due to a bug but looks like it's something that can actually happen.

Reply:
interesting! thx @hrkrshnn  :)